cmd/kueuectl/app/delete/delete_workload.go

```diff
@@ -0,0 +1,386 @@
+
+	for _, name := range o.Names {
+		wl, err := o.Client.Workloads(o.Namespace).Get(ctx, name, metav1.GetOptions{})
+		if client.IgnoreNotFound(err) != nil {
+			return nil, false, err
+		}
+		if err != nil {
+			fmt.Fprintln(o.ErrOut, err)
+			continue
+		}
+
+		workloads = append(workloads, wl)
+		if len(wl.OwnerReferences) > 0 {
+			haveAssociatedWorkloads = true
+		}
+	}
+
+	return workloads, haveAssociatedWorkloads, nil
+}
+
+func (o *WorkloadOptions) getWorkloadResources(workloads []*v1beta1.Workload) (map[*v1beta1.Workload][]GroupVersionResourceName, error) {
+	workloadResources := make(map[*v1beta1.Workload][]GroupVersionResourceName, len(workloads))
+
+	for _, wl := range workloads {
+		workloadResources[wl] = make([]GroupVersionResourceName, 0, len(wl.OwnerReferences))
+
+		for _, or := range wl.OwnerReferences {
+			gv, err := schema.ParseGroupVersion(or.APIVersion)
+			if err != nil {
+				return nil, err
+			}
+
+			mapping, err := o.RestMapper.RESTMapping(gv.WithKind(or.Kind).GroupKind(), gv.Version)
+			if err != nil {
+				return nil, err
+			}
+
+			workloadResources[wl] = append(workloadResources[wl], GroupVersionResourceWithName(mapping.Resource, or.Name))
+		}
+	}
+
+	return workloadResources, nil
+}
+
+func (o *WorkloadOptions) generateConfirmationMessage(workloadResources map[*v1beta1.Workload][]GroupVersionResourceName) string {
+	associatedResources := make([]string, 0, len(workloadResources))
+
+	for wl, resources := range workloadResources {
+		names := make([]string, 0, len(resources))
+		for _, resource := range resources {
+			names = append(names, resource.String())
+		}
+
+		if len(names) > 0 {
+			associatedResource := fmt.Sprintf("  - %s associated with the %s/%s workload\n", strings.Join(names, ", "), wl.Namespace, wl.Name)
+			associatedResources = append(associatedResources, associatedResource)
+		}
+	}
+
+	if len(associatedResources) == 0 {
+		return ""
+	}
+
+	return fmt.Sprintf("This operation will also delete:\n%s", strings.Join(associatedResources, "\n"))
+}
+
+func (o *WorkloadOptions) confirmation(message string) bool {
+	fmt.Fprint(o.Out, message, "Do you want to proceed (y/n)? ")
+
+	var input string
+
+	_, err := fmt.Fscan(o.In, &input)
+	if err != nil {
+		return false
+	}
+
+	return strings.EqualFold(input, "y")
+}
+
+func (o *WorkloadOptions) deleteWorkloads(ctx context.Context, workloadNameResources map[*v1beta1.Workload][]GroupVersionResourceName) error {
+	for wl, nrs := range workloadNameResources {
+		deleteOptions := metav1.DeleteOptions{
+			PropagationPolicy: ptr.To(o.CascadeStrategy),
+		}
+
+		if o.DryRunStrategy == util.DryRunServer {
+			deleteOptions.DryRun = []string{metav1.DryRunAll}
+		}
+
+		for _, nr := range nrs {
+			if o.DryRunStrategy != util.DryRunClient {
+				if err := o.DynamicClient.Resource(nr.GroupVersionResource).Namespace(wl.Namespace).
+					Delete(ctx, nr.Name, deleteOptions); client.IgnoreNotFound(err) != nil {
```

Comment:
If deleteOptions has `Orphan`  PropagationPolicy, the workload will not be deleted as described in the command's description.

Reply:
The cascade option for the delete Workload mixes two layers of abstraction: on one hand we use Workload as a concept representing computation, on the other hand we use the technical option `--cascade` referring to ownerRaference mangement. 

So, I would suggest not supporting the `cascade` for deleting Workloads, and to use the default delete strategy (Background) for Jobs. This would be consistent with the KEP, and be sufficient until we have some use cases for supporting the cascade.

Reply:
Done